cli/tests/activate-mode.bats

```diff
@@ -132,19 +132,20 @@ function assert_run_mode() {
   _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/almonds.yaml" "$FLOX_BIN" install almonds
   # `almonds` brings in Python as a development dependency, and we don't want
   # that in runtime mode
-  run "$FLOX_BIN" activate -m run -- bash <(cat <<'EOF'
+  SCRIPT="$(cat << 'EOF'
     [ -e "$FLOX_ENV/bin/almonds" ]
     [ ! -e "$FLOX_ENV/bin/python3" ]
 EOF
-)
+  )"
+  run "$FLOX_BIN" activate -m run -c "bash -c '$SCRIPT'"
```

Comment:
> - In some places I dropped an extra bash -c since flox activate -c already runs $userShell -c

Isn't `userShell` already `bash` for most of these and if it's not then we'd want to test the shell specified by `FLOX_SHELL`?